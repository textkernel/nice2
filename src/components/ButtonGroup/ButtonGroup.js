import React from 'react';
import PropTypes from 'prop-types';
import bem from '../../packages/bem';
import styles from './ButtonGroup.scss';
import { SIZES } from '../../constants';

const ButtonGroup = ({ children, size, style, block, elem }) => (
    <div
        {...block()}
        style={style}
    >
        { React.Children.map(children, button => (
            React.cloneElement(button, {
                ...button.props,
                size,
                ...elem('Button')
            })
        )) }
    </div>
);

ButtonGroup.propTypes = {
    /** The buttons in this group */
    children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
    /** The size of the button */
    size: PropTypes.oneOf(SIZES),
    /** whether or not to show block - level button(full width) */
    isBlock: PropTypes.bool,
    /** inline styles */
    style: PropTypes.object, // eslint-disable-line react/forbid-prop-types
    /** additional class names */
    className: PropTypes.string
};

ButtonGroup.defaultProps = {
    size: 'normal',
    isBlock: false,
    style: null,
    className: ''
};

ButtonGroup.propsToMods = ['size', 'isBlock'];

export default bem(styles)(ButtonGroup);
