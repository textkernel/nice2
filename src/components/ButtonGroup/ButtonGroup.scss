@import '../../themes/oneui/oneui.scss';

.ButtonGroup {
    display: inline-flex;

    &__button {
        border-radius: 0;

        &--first {
            border-top-left-radius: var(--border-radius);
            border-bottom-left-radius: var(--border-radius);
        }

        &--last {
            border-top-right-radius: var(--border-radius);
            border-bottom-right-radius: var(--border-radius);
        }

        ~ .ButtonGroup__button {
            margin-left: -1px;
        }

        &--context {
            @each $context in ($contexts) {
                &_#{$context} {
                    &:not(.ButtonGroup__button--first):not(:focus) {
                        border-left-color: var(--color-#{$context}-25);
                    }
                    &:not(.ButtonGroup__button--last):not(:focus) {
                        border-right-color: var(--color-#{$context}-25);
                    }
                }
            }
        }
    }

    &--isBlock {
        display: flex;
        width: 100%;
    }

    &__button--isBlock {
        flex-grow: 1;
    }
}
