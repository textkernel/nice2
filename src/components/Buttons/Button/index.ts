export { Button } from './Button';
export { Props as ButtonProps } from './Button';
