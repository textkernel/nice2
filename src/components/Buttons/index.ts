export { Button, ButtonProps } from './Button';
export { SearchButton } from './SearchButton';
