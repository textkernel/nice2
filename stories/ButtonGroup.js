import React from 'react';
import { storiesOf } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
import { boolean, select, withKnobs } from '@storybook/addon-knobs'; // eslint-disable-line import/no-extraneous-dependencies
import ButtonGroup from '../src/components/ButtonGroup';
import Button from '../src/components/Button';
import { SIZES } from '../src/constants';

storiesOf('ButtonGroup', module)
    .addDecorator(withKnobs)
    .add('ButtonGroup', () => (
        <ButtonGroup isBlock={boolean('isBlock', false)} size={select('Size', SIZES, SIZES[1])}>
            <Button context="neutral">Some</Button>
            <Button context="neutral">Button</Button>
            <Button context="neutral">Group</Button>
        </ButtonGroup>
    ));
