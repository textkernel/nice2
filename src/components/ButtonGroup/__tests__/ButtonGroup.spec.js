import React from 'react';
import toJson from 'enzyme-to-json';
import { ButtonGroup } from '../ButtonGroup';
import { Button } from '../../Buttons';

describe('<ButtonGroup> that renders a button', () => {
    it('should render default button correctly', () => {
        const wrapper = mount(
            <ButtonGroup>
                <Button>A button</Button>
                <Button>Another button</Button>
            </ButtonGroup>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should add classes when props are changed', () => {
        const wrapper = mount(
            <ButtonGroup size="large" isBlock>
                <Button>A button</Button>
                <Button>Another button</Button>
            </ButtonGroup>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
