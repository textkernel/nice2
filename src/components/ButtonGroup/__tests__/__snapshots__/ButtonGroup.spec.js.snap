// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonGroup> that renders a button should add classes when props are changed 1`] = `
<ButtonGroup
  isBlock={true}
  isInline={false}
  size="large"
>
  <div
    className="ButtonGroup ButtonGroup--isBlock"
    role="group"
  >
    <Button
      className="ButtonGroup__button ButtonGroup__button--isBlock"
      context="neutral"
      disabled={false}
      href={null}
      isBlock={false}
      isInline={false}
      key=".0"
      size="large"
      type="button"
    >
      <button
        className="Button Button--context_neutral Button--size_large ButtonGroup__button ButtonGroup__button--isBlock"
        disabled={false}
        type="button"
      >
        A button
      </button>
    </Button>
    <Button
      className="ButtonGroup__button ButtonGroup__button--isBlock"
      context="neutral"
      disabled={false}
      href={null}
      isBlock={false}
      isInline={false}
      key=".1"
      size="large"
      type="button"
    >
      <button
        className="Button Button--context_neutral Button--size_large ButtonGroup__button ButtonGroup__button--isBlock"
        disabled={false}
        type="button"
      >
        Another button
      </button>
    </Button>
  </div>
</ButtonGroup>
`;

exports[`<ButtonGroup> that renders a button should render default button correctly 1`] = `
<ButtonGroup
  isBlock={false}
  isInline={false}
  size="normal"
>
  <div
    className="ButtonGroup"
    role="group"
  >
    <Button
      className="ButtonGroup__button"
      context="neutral"
      disabled={false}
      href={null}
      isBlock={false}
      isInline={false}
      key=".0"
      size="normal"
      type="button"
    >
      <button
        className="Button Button--context_neutral ButtonGroup__button"
        disabled={false}
        type="button"
      >
        A button
      </button>
    </Button>
    <Button
      className="ButtonGroup__button"
      context="neutral"
      disabled={false}
      href={null}
      isBlock={false}
      isInline={false}
      key=".1"
      size="normal"
      type="button"
    >
      <button
        className="Button Button--context_neutral ButtonGroup__button"
        disabled={false}
        type="button"
      >
        Another button
      </button>
    </Button>
  </div>
</ButtonGroup>
`;
